data/transactions/logic/eval_test.go

```diff
@@ -3187,31 +2819,161 @@ func benchmarkBasicProgram(b *testing.B, source string) {
+		{"b^", "byte 0x01234576",  "byte 0x0223627389; b^", "pop; int 1"},
+		{"b~", "byte 0x0123457673624736", "b~", "pop; int 1"},
+
+		{"b&big",
+			"byte 0x0123457601234576012345760123457601234576012345760123457601234576",
+			"byte 0x01ffffffffffffff01ffffffffffffff01234576012345760123457601234576; b&",
+			"pop; int 1"},
+		{"b|big",
+			"byte 0x0123457601234576012345760123457601234576012345760123457601234576",
+			"byte           0xffffff01ffffffffffffff01234576012345760123457601234576; b|",
+			"pop; int 1"},
+		{"b^big", "",	// u256*u256
+			`byte 0x123457601234576012345760123457601234576012345760123457601234576a
+			 byte 0xf123457601234576012345760123457601234576012345760123457601234576; b^; pop`,
+			"int 1"},
+		{"b~big", "byte 0xa123457601234576012345760123457601234576012345760123457601234576",
+			"b~",
+			"pop; int 1"},
+	}
+	for _, bench := range benches {
+		b.Run(bench[0], func(b *testing.B) {
+			benchmarkOperation(b, bench[1], bench[2], bench[3]);
+		})
+	}
+}
+
+func BenchmarkBigMath(b *testing.B) {
+	benches := [][]string{
+		{"bpop", "", "byte 0x01ffffffffffffff; pop", "int 1"},
+
+		{"b+", "byte 0x01234576", "byte 0x01ffffffffffffff; b+", "pop; int 1"},
+		{"b-", "byte 0x0ffff1234576", "byte 0x1202; b-", "pop; int 1"},
+		{"b*", "", "byte 0x01234576; byte 0x0223627389; b*; pop", "int 1"},
+		{"b/", "", "byte 0x0123457673624736; byte 0x0223627389; b/; pop", "int 1"},
+		{"b%", "", "byte 0x0123457673624736; byte 0x0223627389; b/; pop", "int 1"},
+
+		{"b+big",	// u256 + u256
+			"byte 0x0123457601234576012345760123457601234576012345760123457601234576",
+			"byte 0x01ffffffffffffff01ffffffffffffff01234576012345760123457601234576; b+",
+			"pop; int 1"},
+		{"b-big",	// second is a bit small, so we can subtract it over and over
+			"byte 0x0123457601234576012345760123457601234576012345760123457601234576",
+			"byte           0xffffff01ffffffffffffff01234576012345760123457601234576; b-",
+			"pop; int 1"},
+		{"b*big", "",	// u256*u256
+			`byte 0xa123457601234576012345760123457601234576012345760123457601234576
+			 byte 0xf123457601234576012345760123457601234576012345760123457601234576; b*; pop`,
+			"int 1"},
+		{"b/big", "",	// u256 / u128 (half sized divisor seems pessimal)
+			`byte 0xa123457601234576012345760123457601234576012345760123457601234576
+			 byte 0x34576012345760123457601234576312; b/; pop`,
+			"int 1"},
+		{"b%big", "",	// u256 / u128 (half sized divisor seems pessimal)
+			`byte 0xa123457601234576012345760123457601234576012345760123457601234576
+			 byte 0x34576012345760123457601234576312; b/; pop`,
+			"int 1"},
+	}
+	for _, bench := range benches {
+		b.Run(bench[0], func(b *testing.B) {
+			benchmarkOperation(b, bench[1], bench[2], bench[3]);
+		})
+	}
+}
+
+func BenchmarkHash(b *testing.B) {
+	hashes := []string{"sha256", "keccak256", "sha512_256"}
+	for _, hash := range hashes {
+		b.Run(hash+"-small", func(b *testing.B) { // hash 32 bytes
+			benchmarkOperation(b, "byte 0x1234567890", hash,
```

Comment:
nit: for each of these its only hashing 5 bytes on the first execution, and then {32,128,512} for the next 1999. It could be perfectly consistent by using a 32 byte constant and running `dup; concat; dup; concat; {hash}` as the repeated op?